[WIP] OSDOCS:2475 Adds new assembly for the dynamic plugin TP (#41260)
openshift/openshift-docs: modules/about-dynamic-plug-ins.adoc

// Module is included in the following assemblies:
//
// * openshift-docs/web_console/dynamic-plug-ins.adoc

:_content-type: CONCEPT
[id="about-dynamic-plugins"]
= About dynamic plug-ins

> The dynamic plug-in allows you to add custom pages and other extensions to your interface at runtime. Plug-ins are registered with the console using the `ConsolePlugin` custom resource, and they are enabled in the console operator configuration by a cluster administrator.

Reviewer: "Dynamic plugins" or "A dynamic plug-in"? "The dynamic plug-in" implies that there is only one.

[id="dynamic-plugins-features"]
== Key features

> The dynamic plug-in will allow you to make customizations such as:

Reviewer: "Dynamic plugins" or "A dynamic plug-in"?

> * Add custom pages and other extensions

Reviewer: These are all extensions, so I'd remove "and other extensions."

```suggestion
* Add custom pages
```

// Module is included in the following assemblies:
//
// * openshift-docs/web_console/dynamic-plug-ins.adoc

:_content-type: CONCEPT
[id="about-dynamic-plugins_{context}"]
= About dynamic plug-ins

> A dynamic plug-in allows you to add custom pages and other extensions to your interface at runtime. Plug-ins are registered with the console using the `ConsolePlugin` custom resource, and they are enabled in the `console-operator` configuration by a cluster administrator.

Reviewer 2: ```suggestion
A dynamic plug-in allows you to add custom pages and other extensions to your interface at runtime. The `ConsolePlugin` custom resource registers plug-ins with the console, and a cluster administrator enables plug-ins in the `console-operator` configuration.
```
Just a suggestion if you want to avoid passive voice.

[id="dynamic-plugins-features"]
== Key features

A dynamic plug-in allows you to make customizations to the {product-title} experience.

* Add custom pages.
* Add perspectives and update navigation items.
* Add tabs and actions to resource pages.

> == PatternFly 4 Guidelines

Reviewer 2: ```suggestion
== PatternFly 4 guidelines
```

When creating your plug-in, follow these guidelines for using PatternFly:

* Use link:https://www.patternfly.org/v4/[PatternFly4] components and PatternFly CSS variables. Core PatternFly components are available through the SDK. Using PatternFly components and variables will help your plug-in look consistent in future console versions.
* Make your plug-in accessible by following link:https://www.patternfly.org/v4/accessibility/accessibility-fundamentals/[PatternFly's accessibility fundamentals].
* _Do not_ use other CSS libraries such as Bootstrap or Tailwind. They can conflict with PatternFly and will not match the console look and feel.

[id="general-plug-in-guidelines"]

> == General Guidelines

Reviewer 2: ```suggestion
== General guidelines
```

When creating your plug-in, follow these general guidelines:

> * Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin__heading` and `my-plugin__icon`.

Reviewer: I'm not sure the right asciidoc syntax to escape the `__`. It's still showing up as italics. I think perhaps `\`? (untested)

```suggestion
* Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin\_\_heading` and `my-plugin\_\_icon`.
```

Author: Looks like that worked. I will have some one on our peer review rotation check it. Thanks!

// Module is included in the following assemblies:
//
// * openshift-docs/web_console/dynamic-plug-ins.adoc

:_content-type: CONCEPT
[id="about-dynamic-plugins_{context}"]
= About dynamic plug-ins

A dynamic plug-in allows you to add custom pages and other extensions to your interface at runtime. Plug-ins are registered with the console using the `ConsolePlugin` custom resource, and they are enabled in the console operator configuration by a cluster administrator.

[id="dynamic-plugins-features"]
== Key features

A dynamic plug-in will allow you to make customizations such as:

* Add custom pages
* Add perspectives and update navigation items
* Add tabs and actions to resource pages

== PatternFly 4 Guidelines
When creating your plug-in, follow these guidelines for using PatternFly:

* Use link:https://www.patternfly.org/v4/[PatternFly4] components. Core PatternFly components are available through the SDK.

> * Use PatternFly components and PatternFly CSS variables. Using PatternFly components and variables will help your plug-in look consistent in future console versions.

Reviewer: Should we combine these two bullets? (Since they both say to use PF components?)

> * Make your plug-in accessible by using link:https://www.patternfly.org/v4/accessibility/accessibility-fundamentals/[PatternFly's accessibility fundamentals].

Reviewer: ```suggestion
* Make your plug-in accessible by following link:https://www.patternfly.org/v4/accessibility/accessibility-fundamentals/[PatternFly's accessibility fundamentals].
```

* _Do not_ use other CCS libraries such as Bootstrap or Tailwind. They can conflict with PatternFly and will not match the console look and feel.

[id="general-plug-in-guidelines"]
== General Guidelines
When creating your plug-in, follow these general guidelines:

> * Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin__heading, my-plugin__icon`

Reviewer: It looks like the `__` is causing italics text in the generated content. Maybe this fixes it?

```suggestion
* Prefix your CSS class names with your plug-in name to avoid collisions. For example, `my-plugin__heading` and `my-plugin__icon`
```
